File: src/main/java/org/zbus/common/Proto.java
package org.zbus.common;

import java.util.Map;

import org.remoting.Message;

public class Proto {
	//生产消费者模式
	public static final String Produce   = "produce";     //生产消息
	public static final String Consume   = "consume";     //消费消息 
	
	//请求回复模式: 1个生产消费队列 + n个临时回复队列
	public static final String Request   = "request";     //请求等待应答消息 
	
	
	//心跳
	public static final String Heartbeat = "heartbeat"; //心跳消息
	
	//管理类
	public static final String Admin = "admin";      //管理类消息
	public static final String CreateMQ = "create_mq";      
	
	//TrackServer通讯
	public static final String TrackReport = "track_report"; 
	public static final String TrackSub    = "track_sub";  
	public static final String TrackPub    = "track_pub"; 
	
	public static Message buildAdminMessage(String registerToken, String cmd, Map<String, String> params){
    	Message msg = new Message();
    	msg.setCommand(Admin); 
    	msg.setToken(registerToken);
    	
    	msg.setHead("cmd", cmd); 
    	for(Map.Entry<String, String> e : params.entrySet()){
    		msg.setHead(e.getKey(), e.getValue());
    	}
    	return msg;
    }
	 
}

File: api/csharp/zbus/Proto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using zbus.Remoting;
namespace zbus
{
    public enum MessageMode
    {
        MQ = 1<<0,
        PubSub = 1<<1,
        Temp = 1<<2,
    }

    public class Proto
    {
        public static readonly string Produce     = "produce";    //生产消息
        public static readonly string Consume     = "consume";    //消费消息 
        public static readonly string Request     = "request";    //请求等待应答消息  
        public static readonly string Heartbeat   = "heartbeat";  //心跳消息
        public static readonly string Admin       = "admin";      //管理类消息
        public static readonly string CreateMQ    = "create_mq";      
        public static readonly string TrackReport = "track_report";
        public static readonly string TrackSub    = "track_sub";
        public static readonly string TrackPub    = "track_pub";


        public static Message BuildAdminMessage(string registerToken, string cmd, IDictionary<string, string> args)
        {
            Message msg = new Message();
            msg.Command = Proto.Admin;
            msg.Token = registerToken;
            msg.SetHead("cmd", cmd);
            foreach (KeyValuePair<string,string> kv in args)
            {
                msg.SetHead(kv.Key, kv.Value);
            }
            return msg;
        }
    }
}
